src/hssm/distribution_utils/onnx/onnx.py

```diff
@@ -51,6 +52,63 @@ def make_jax_logp_funcs_from_onnx(
         onnx.load(str(model)) if isinstance(model, (str, PathLike)) else model
     )
 
+    if data_dim == 0:
+
+        def logp_no_data(*dist_params: float) -> float:
+            """Compute the log-likelihood.
+
+            A function that computes the element-wise log-likelihoods given one
+            data point and arbitrary numbers of parameters as scalars.
+
+            Parameters
+            ----------
+            data
+                A 1-D length 2 array with response time and response that
+                represents one data point
+            dist_params
+                A list of parameters used in the likelihood computation.
+
+            Returns
+            -------
+            float
+                The element-wise log-likelihoods.
+            """
+            # Makes a matrix to feed to the LAN model
+            input_vector = jnp.array(dist_params)
+
+            return interpret_onnx(loaded_model.graph, input_vector)[0].squeeze()
+
+        # The vectorization of the logp function
+        vmap_logp_no_data = vmap(
+            logp_no_data,
+            in_axes=[0 if is_regression else None for is_regression in params_is_reg],
+        )
+
+        def vjp_vmap_logp_no_data(
+            *dist_params: list[float | ArrayLike], gz: ArrayLike
+        ) -> list[ArrayLike]:
+            """Compute the VJP of the log-likelihood function.
+
+            Parameters
+            ----------
+            data
+                A two-column numpy array with response time and response.
+            dist_params
+                A list of parameters used in the likelihood computation.
+            gz
+                The value of vmap_logp at which the VJP is evaluated, typically is just
+                vmap_logp(data, *dist_params)
+
+            Returns
+            -------
+            list[ArrayLike]
+                The VJP of the log-likelihood function computed at gz.
+            """
+            _, vjp_fn = vjp(vmap_logp_no_data, *dist_params)
+            return vjp_fn(gz)[1:]
```

Comment:
just as a question. Can't immediately remember why the return here skips `0` index.

Reply:
Here's the Jax Doc

> If has_aux is False, returns a (primals_out, vjpfun) pair, where primals_out is fun(*primals). If has_aux is True, returns a (primals_out, vjpfun, aux) tuple where aux is the auxiliary data returned by fun.

> vjpfun is a function from a cotangent vector with the same shape as primals_out to a tuple of cotangent vectors with the same number and shapes as primals, representing the vector-Jacobian product of fun evaluated at primals.

Looks like we don't need the primals

Reply:
Gotta love how complicated this reads...